File: python/trip_record.py
from dataclasses import dataclass


@dataclass
class TripRecord:
    vendor_id: int
    pickup_timestamp: int
    dropoff_timestamp: int
    passenger_count: int
    trip_distance: float
    rate_code_id: int
    store_and_fwd_flag: bool
    pu_location_id: int
    do_location_id: int
    payment_type: int
    fare_amount: float
    extra: float
    mta_tax: float
    tip_amount: float
    tolls_amount: float
    improvement_surcharge: float
    total_amount: float

File: cpp/src/TripRecord.cpp
#include "taxi/TripRecord.hpp"

namespace taxi {

bool TripRecord::is_valid() const {
    // Basic validation: ensure timestamps are reasonable and positive amounts
    return pickup_timestamp > 0 && 
           dropoff_timestamp > pickup_timestamp &&
           passenger_count >= 0 &&
           trip_distance >= 0.0 &&
           total_amount >= 0.0;
}

} // namespace taxi
